lib/tests/test_core.c

```diff
@@ -48,12 +48,82 @@ test_strerror_tskit(void)
     }
 }
 
+static void
```

Comment:
gee, nice tests here

Reply:
Ah yes, those cases at the end I put in since I thought you might be interested in seeing the case you brought up. You can see how in your case with DBL_EPSILON / 8.0 in the middle a random pick is possible, but then DBL_EPSILON / 16.0 it becomes "very small" in the sense that it is effectively zero and will make that pick impossible.